addon/modifiers/did-resize.js

```diff
@@ -6,23 +6,41 @@ export default class DidResizeModifier extends Modifier {
   options = {};
 
   // Private API
-  observer = null;
+  _didInstall = false;
+  static observer = null;
+  static handlers = [];
+
+  addHandler() {
+    DidResizeModifier.handlers.push({
+      element: this.element,
+      handler: this.handler,
+    });
+  }
+
+  removeHandler() {
+    DidResizeModifier.handlers.splice(
+      DidResizeModifier.handlers.findIndex((h) => h.element === this.element),
+      1
+    );
+  }
 
   observe() {
-    if (this.observer) {
-      this.observer.observe(this.element, this.options);
+    if (DidResizeModifier.observer) {
+      this.addHandler();
+      DidResizeModifier.observer.observe(this.element, this.options);
     }
   }
 
   unobserve() {
-    if (this.observer) {
-      this.observer.unobserve();
+    if (DidResizeModifier.observer) {
+      DidResizeModifier.observer.unobserve(this.element);
```

Comment:
This method should probably also call `removeHandler`, since it's possible that the modifier's `handler` has been changed by the consumer.